templates/CMakeLists.txt.template

```diff
@@ -295,57 +298,53 @@
   add_custom_target(tools
     DEPENDS tools_c tools_cxx)
 
-  if (gRPC_BUILD_TESTS)
-  add_custom_target(buildtests_c)
-  % for tgt in targets:
-  % if tgt.build == 'test' and not tgt.language == 'c++' and not tgt.get('external_deps', None) and not tgt.boringssl:
-  ${get_platforms_condition_begin(tgt.platforms)}\
-  add_dependencies(buildtests_c ${tgt.name})
-  ${get_platforms_condition_end(tgt.platforms)}\
-  % endif
-  % endfor
+  if(gRPC_BUILD_TESTS)
+    add_custom_target(buildtests_c)
+    % for tgt in targets:
+    % if tgt.build == 'test' and not tgt.language == 'c++' and not tgt.get('external_deps', None) and not tgt.boringssl:
+    ${platforms_condition_add_dependencies(tgt, 'buildtests_c')}
```

Comment:
One more issue: I'm not sure if introduction of `platforms_condition_add_dependencies` actually makes things more readable, especially because we had to add whitespaces some of the functions to end up with the correct indentation. Would just
```
${get_platforms_condition_begin(tgt.platforms)}
  add_dependencies(buildtests_c ${tgt.name})
${get_platforms_condition_end(tgt.platforms)}
```
(and removing the \n and spaces from the python functions) work?

Reply:
No :( I tried a couple different approaches to fix the whitespace (including your idea), and this was the only one that worked properly.

Reply:
I just pushed an updated approach using `<%block>` and a filter. PTAL.